kore/src/Kore/Step/Simplification/Predicate.hs

```diff
@@ -52,15 +51,22 @@ simplifyPartial
   = do
     (patternOr, _proof) <-
         Monad.Trans.lift
-        $ simplifier substitutionSimplifier (unwrapPredicate predicate)
+        $ simplifier substitutionSimplifier
+        $ Predicate.unwrapPredicate predicate
     -- Despite using Monad.Trans.lift above, we do not need to explicitly check
     -- for \bottom because patternOr is an OrOfExpandedPattern.
-    scatter (eraseTerm <$> patternOr)
+    scatter (rewrapPredicate <$> patternOr)
   where
-    eraseTerm predicated@Predicated { term }
-      | Top_ _ <- term = predicated { term = () }
-      | otherwise =
-        (error . show . Pretty.vsep)
-            [ "Expecting a \\top term, but found:"
-            , unparse predicated
-            ]
+    rewrapPredicate predicated =
+        predicated
+            { term = ()
+            , predicate =
+                -- It is safe to re-wrap the simplified term as a predicate:
+                -- simplifying a predicate must produce a predicate because
+                -- simplification preserves sorts.
+                Predicate.makeAndPredicate
+                    (Predicate.wrapPredicate simplified)
```

Comment:
It should be wrong to wrap `simplified`. If it is really a predicate, simplification should have already put it in the `predicate` field. Why does this make sense?

Reply:
> If it is really a predicate, simplification should have already put it in the predicate field.
That is true only if it is something we know how to simplify, but there are patterns which are not handled, such as `\not(...)`.

Reply:
That should _always_ be true, including not-handled patterns like not. Simplification over, say, a `not (a == b)` should identify that `a == b` is a predicate, and applying `not` on top of it should not change that. Simplification over `(not a) == b` should also identify that `==` produces predicates.

The only case in which this does not apply is for things like `a or (not a)` where `a` is not a predicate, but, if we have those as predicates, we should just fail here since those are not handled properly by most of our code, including the SMT.

@traiansf since I may be wrong about this.

BTW, that's why the documentation for `wrapPredicate` says `This is intended for predicate evaluation and tests and should not be used outside of that.` The message could have been better.